src/app/portlets/dot-edit-page/layout/dot-edit-layout-grid/dot-edit-layout-grid.component.ts

```diff
@@ -119,59 +91,53 @@ export class DotEditLayoutGridComponent implements OnInit {
     }
 
     /**
-     * Event fired when the grad of a container ends, remove empty rows if any.
-     * @constructor
+     * Event fired when the drag of a container ends, remove empty rows if any.
+     *
      */
     onDragStop(): void {
         this.deleteEmptyRows();
     }
 
     /**
-     * Transform the grid data into DotLayoutBody.
-     * @returns {DotLayoutBody}
+     * Return ng-grid model.
+     * @returns {DotLayoutGridBox[]}
      */
-    getLayoutBody(): DotLayoutBody {
-        return this.dotEditLayoutGridService.getDotLayoutBody(this.gridBoxes);
+    getGrid(): DotLayoutGridBox[] {
+        return this.grid;
     }
 
     private removeContainer(index: number): void {
-        if (this.gridBoxes[index]) {
-            this.gridBoxes.splice(index, 1);
+        if (this.grid[index]) {
+            this.grid.splice(index, 1);
             this.deleteEmptyRows();
         }
     }
 
     private setConfigOfNewContainer(): NgGridItemConfig {
-        let lastContainer;
-        const newRow: NgGridItemConfig = Object.assign({}, DotEditLayoutGridComponent.NEW_ROW_TEMPLATE);
-        let busyColumns: number = DotEditLayoutGridComponent.NEW_ROW_TEMPLATE.sizex;
-        if (this.gridBoxes.length) {
-            // check last row && last column in last row
-            lastContainer = this.gridBoxes.reduce(
-                (currentContainer: DotLayoutGridBox, nextContainer: DotLayoutGridBox) => {
-                    return currentContainer.config.row > currentContainer.config.row
-                        ? currentContainer
-                        : currentContainer.config.row === nextContainer.config.row
-                          ? currentContainer.config.col > nextContainer.config.col ? currentContainer : nextContainer
-                          : nextContainer;
-                }
-            );
+        const newRow: NgGridItemConfig = Object.assign({}, DOT_LAYOUT_GRID_NEW_ROW_TEMPLATE);
+
+        if (this.grid.length) {
+            const lastContainer = _.last(_.sortBy(this.grid, 'config.row'));
```

Comment:
When you have to transform or get something out from a big object, check lodash documentation, you'll be surprise.